UEBPrep/app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -10,6 +10,17 @@ def google_oauth2
       session["devise.google_data"] = request.env["omniauth.auth"]
       redirect_to new_user_registration_url
     end
+  end
 
+  def facebook
+    @user = User.from_omniauth(request.env["omniauth.auth"])
```

Comment:
This works now, having one single from_omniauth model method to handle both cases of FB and Google+, if Twitter ends up having a different response setup, we'll have to customize one for Twiiter.